File: src/main/java/fr/stromans/service/dto/ClassificationRuleResultDto.java
package fr.stromans.service.dto;

import fr.stromans.domain.Transaction;

import java.util.ArrayList;
import java.util.List;

public class ClassificationRuleResultDto {

    private List<Transaction> transactionsToClassify = new ArrayList<>();

    public ClassificationRuleResultDto() {
        // Empty constructor needed for Jackson.
    }

    public List<Transaction> getTransactionsToClassify() {
        return transactionsToClassify;
    }

    public void setTransactionsToClassify(List<Transaction> transactionsToClassify) {
        this.transactionsToClassify = transactionsToClassify;
    }

    public void appendIgnoredTransaction(Transaction transaction){
        this.transactionsToClassify.add(transaction);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "ClassificationRuleResultDto{" +
            ", transactionsToClassify=" + transactionsToClassify +
            "}";
    }
}

File: src/main/webapp/app/shared/model/dto/classification-rule-result-dto.ts
import { ITransaction } from '../transaction.model';

export interface IClassificationRuleResultDto {
  transactionsToClassify: ITransaction[];
}

export class ClassificationRuleResultDto implements IClassificationRuleResultDto {
  constructor(public transactionsToClassify: ITransaction[]) {
    this.transactionsToClassify = transactionsToClassify || [];
  }
}
